server/lib/pbench/process_tb.py

```diff
@@ -49,7 +50,7 @@ def _get_receive_dir(self) -> Path:
         return receive_dir
 
     @staticmethod
-    def _results_push(tb: Path, token: str, satellite: str = None):
+    def _results_push(tb: Path, token: str, satellite: Optional[str]):
```

Comment:
Don't you still need to make it optional by giving it a default value of `None`?

Reply:
Isn't it optional because it can be `None` or `str`?

Reply:
`satellite: Optional[str]` tells *humans* and analysis tools that the *required* value of the parameter can be `str` or `None`.

`satellite = None` tells *Python* that if the parameter is omitted, the method runs as if you'd given the value `None`. It's *optional* in the call syntax.

`satellite: str = None` does the same thing, but it's an incorrect type hint as a value of `None` is not a `str`. That's why the ultimate correct syntax is `satellite: Optional[str] = None`.

But if we're happy to always require passing a value (either a `str` or `None`) without the *default*, then `satellite: Optional[str]` is good.